sql/src/main/java/org/apache/druid/sql/http/SqlQuery.java

```diff
@@ -343,5 +341,8 @@ private static SqlQuery from(
     catch (IOException e) {
       throw new HttpException(Response.Status.BAD_REQUEST, "Unable to read query from request: " + e.getMessage());
     }
+    catch (IllegalArgumentException e) {
+      throw new HttpException(Response.Status.BAD_REQUEST, "Invalid Content-Type header: " + e.getMessage());
```

Comment:
Not sure if this is right for this location. There's a lot going on in the `try` block and maybe something else can throw `IllegalArgumentException` other than an invalid content-type. Why not throw this exception directly in the `else` block of the content-type checking conditional?

Reply:
Thanks for the catch. Fixed it. The earlier catch location was an attempt to make switch statement work until I dropped that idea in the final cut .